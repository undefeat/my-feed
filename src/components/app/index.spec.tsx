import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import App from '.';

it('renders with "No Posts Found"', () => {
    const rendered = createRenderer(),
        tree = rendered.render(<App users={[]} posts={[]} loading={false} initialScrollTop={0} />);

    expect(tree).toMatchSnapshot();
});

it('renders with loader', () => {
    const rendered = createRenderer(),
        tree = rendered.render(<App users={[]} posts={[]} loading={true} initialScrollTop={0} />);

    expect(tree).toMatchSnapshot();
});

it('renders with post list', () => {
    const users = [
            { id: 1, name: 'Leanne Graham' },
            { id: 2, name: 'Ervin Howell' },
            { id: 3, name: 'Clementine Bauch' },
            { id: 4, name: 'Patricia Lebsack' },
            { id: 5, name: 'Chelsey Dietrich' },
            { id: 6, name: 'Mrs. Dennis Schulist' },
            { id: 7, name: 'Kurtis Weissnat' },
            { id: 8, name: 'Nicholas Runolfsdottir V' },
            { id: 9, name: 'Glenna Reichert' },
            { id: 10, name: 'Clementina DuBuque' }
        ],
        posts = [
            {
                userId: 1,
                id: 1,
                title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
                body:
                    'quia et suscipit\nsuscipit recusandae consequuntur expedita et cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet architecto'
            },
            {
                userId: 1,
                id: 2,
                title: 'qui est esse',
                body:
                    'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla'
            },
            {
                userId: 1,
                id: 3,
                title: 'ea molestias quasi exercitationem repellat qui ipsa sit aut',
                body:
                    'et iusto sed quo iure\nvoluptatem occaecati omnis eligendi aut ad\nvoluptatem doloribus vel accusantium quis pariatur\nmolestiae porro eius odio et labore et velit aut'
            },
            {
                userId: 1,
                id: 4,
                title: 'eum et est occaecati',
                body:
                    'ullam et saepe reiciendis voluptatem adipisci\nsit amet autem assumenda provident rerum culpa\nquis hic commodi nesciunt rem tenetur doloremque ipsam iure\nquis sunt voluptatem rerum illo velit'
            },
            {
                userId: 1,
                id: 5,
                title: 'nesciunt quas odio',
                body:
                    'repudiandae veniam quaerat sunt sed\nalias aut fugiat sit autem sed est\nvoluptatem omnis possimus esse voluptatibus quis\nest aut tenetur dolor neque'
            },
            {
                userId: 1,
                id: 6,
                title: 'dolorem eum magni eos aperiam quia',
                body:
                    'ut aspernatur corporis harum nihil quis provident sequi\nmollitia nobis aliquid molestiae\nperspiciatis et ea nemo ab reprehenderit accusantium quas\nvoluptate dolores velit et doloremque molestiae'
            },
            {
                userId: 1,
                id: 7,
                title: 'magnam facilis autem',
                body:
                    'dolore placeat quibusdam ea quo vitae\nmagni quis enim qui quis quo nemo aut saepe\nquidem repellat excepturi ut quia\nsunt ut sequi eos ea sed quas'
            },
            {
                userId: 1,
                id: 8,
                title: 'dolorem dolore est ipsam',
                body:
                    'dignissimos aperiam dolorem qui eum\nfacilis quibusdam animi sint suscipit qui sint possimus cum\nquaerat magni maiores excepturi\nipsam ut commodi dolor voluptatum modi aut vitae'
            },
            {
                userId: 1,
                id: 9,
                title: 'nesciunt iure omnis dolorem tempora et accusantium',
                body:
                    'consectetur animi nesciunt iure dolore\nenim quia ad\nveniam autem ut quam aut nobis\net est aut quod aut provident voluptas autem voluptas'
            },
            {
                userId: 1,
                id: 10,
                title: 'optio molestias id quia eum',
                body:
                    'quo et expedita modi cum officia vel magni\ndoloribus qui repudiandae\nvero nisi sit\nquos veniam quod sed accusamus veritatis error'
            }
        ];

    const rendered = createRenderer(),
        tree = rendered.render(<App users={users} posts={posts} loading={false} initialScrollTop={0} />);

    expect(tree).toMatchSnapshot();
});
