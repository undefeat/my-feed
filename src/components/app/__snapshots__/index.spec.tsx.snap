// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with "No Posts Found" 1`] = `
<main
  className="app"
>
  <h1>
    Recent posts
  </h1>
  <h2>
    No Posts Found
  </h2>
</main>
`;

exports[`renders with loader 1`] = `
<main
  className="app"
>
  <h1>
    Recent posts
  </h1>
  <Loader />
</main>
`;

exports[`renders with post list 1`] = `
<main
  className="app"
>
  <h1>
    Recent posts
  </h1>
  <PostListContainer
    initialScrollTop={0}
    onScrollTopChanged={[Function]}
    overscan={10}
    postInfos={
      Array [
        Object {
          "author": "Leanne Graham",
          "body": "quia et suscipit
      suscipit recusandae consequuntur expedita et cum
      reprehenderit molestiae ut ut quas totam
      nostrum rerum est autem sunt rem eveniet architecto",
          "id": 1,
          "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        },
        Object {
          "author": "Leanne Graham",
          "body": "est rerum tempore vitae
      sequi sint nihil reprehenderit dolor beatae ea dolores neque
      fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
      qui aperiam non debitis possimus qui neque nisi nulla",
          "id": 2,
          "title": "qui est esse",
        },
        Object {
          "author": "Leanne Graham",
          "body": "et iusto sed quo iure
      voluptatem occaecati omnis eligendi aut ad
      voluptatem doloribus vel accusantium quis pariatur
      molestiae porro eius odio et labore et velit aut",
          "id": 3,
          "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
        },
        Object {
          "author": "Leanne Graham",
          "body": "ullam et saepe reiciendis voluptatem adipisci
      sit amet autem assumenda provident rerum culpa
      quis hic commodi nesciunt rem tenetur doloremque ipsam iure
      quis sunt voluptatem rerum illo velit",
          "id": 4,
          "title": "eum et est occaecati",
        },
        Object {
          "author": "Leanne Graham",
          "body": "repudiandae veniam quaerat sunt sed
      alias aut fugiat sit autem sed est
      voluptatem omnis possimus esse voluptatibus quis
      est aut tenetur dolor neque",
          "id": 5,
          "title": "nesciunt quas odio",
        },
        Object {
          "author": "Leanne Graham",
          "body": "ut aspernatur corporis harum nihil quis provident sequi
      mollitia nobis aliquid molestiae
      perspiciatis et ea nemo ab reprehenderit accusantium quas
      voluptate dolores velit et doloremque molestiae",
          "id": 6,
          "title": "dolorem eum magni eos aperiam quia",
        },
        Object {
          "author": "Leanne Graham",
          "body": "dolore placeat quibusdam ea quo vitae
      magni quis enim qui quis quo nemo aut saepe
      quidem repellat excepturi ut quia
      sunt ut sequi eos ea sed quas",
          "id": 7,
          "title": "magnam facilis autem",
        },
        Object {
          "author": "Leanne Graham",
          "body": "dignissimos aperiam dolorem qui eum
      facilis quibusdam animi sint suscipit qui sint possimus cum
      quaerat magni maiores excepturi
      ipsam ut commodi dolor voluptatum modi aut vitae",
          "id": 8,
          "title": "dolorem dolore est ipsam",
        },
        Object {
          "author": "Leanne Graham",
          "body": "consectetur animi nesciunt iure dolore
      enim quia ad
      veniam autem ut quam aut nobis
      et est aut quod aut provident voluptas autem voluptas",
          "id": 9,
          "title": "nesciunt iure omnis dolorem tempora et accusantium",
        },
        Object {
          "author": "Leanne Graham",
          "body": "quo et expedita modi cum officia vel magni
      doloribus qui repudiandae
      vero nisi sit
      quos veniam quod sed accusamus veritatis error",
          "id": 10,
          "title": "optio molestias id quia eum",
        },
      ]
    }
    rowCount={10}
  />
</main>
`;
